### DISCUSSION TITLE
Async/Await within WebSocketBehavior message handler

### DISCUSSION THREAD

**Discussion by a-deal** [uNetworking/uWebSockets.js#578] [Q&A] [answered]
Hi!

It doesn't seem like async / await is supported from within the `WebSocketBehavior` message handler, but promises/callbacks are. Is this by design? Would it be possible to include this in a future roadmap milestone?

Thanks in advance :-)

**Comment by ghost** [comment, marked as answer]
You can have async handlers?

**Reply by a-deal** [reply]
Guess I'm confused by this description?

`Handler for a WebSocket message. Messages are given as ArrayBuffer no matter if they are binary or not. Given ArrayBuffer is valid during the lifetime of this callback (until first await or return) and will be neutered.`

**Reply by ghost** [reply]
You can have async handlers but the message does not survive first await or return. Copy the message if you want to keep it.

**Reply by a-deal** [reply]
Will do :+1:
